client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -48,6 +48,16 @@ const PageViewTrackerWrapper = ( { category, selectedSiteId, trackPageViews } )
 };
 
 const PluginsBrowser = ( { trackPageViews = true, category, search, hideHeader } ) => {
+	useEffect( () => {
+		switch ( category ) {
+			// Scroll back to top if user is coming from "Discover".
+			case 'popular':
+			case 'featured':
+			case 'paid':
+				window.scroll( 0, 0 );
+		}
+	}, [ category ] );
```

Comment:
Will this scroll to top on every category click or just the first?

Not totally sure but potentially want:

```suggestion
	}, [] );
```
We should aim to match production's current behavior.

Reply:
> Will this scroll to top on every category click or just the first?

Will only scroll to the top when the categories are `paid`, and `popular`(I'll remove `featured` since the Discover page doesn't link to it). They are accessed from the "Discover" page. So if we don't scroll to top, the behavior would be a bit weird from the user's perspective (also different from production).

Screencast if we **apply the suggested change**:

https://user-images.githubusercontent.com/6586048/186809014-060e9933-fd9d-4e2a-bcac-2e1a88ee229b.mp4

> We should aim to match production's current behavior.

The current behavior on **production**:

https://user-images.githubusercontent.com/6586048/186806456-bbbe740c-1937-4d2d-a290-9d2c1a27e101.mp4

When `category` is part of `useEffect`'s dependencies:

https://user-images.githubusercontent.com/6586048/186809161-82cbf9d7-ff6e-45a0-82b7-897ea3e5ecc5.mp4

> Will this scroll to top on every category click or just the first?

For other categories, they won't be scrolled back to the top because it doesn't fulfill the switch cases:

https://user-images.githubusercontent.com/6586048/186809376-bd0b782c-6d3a-4212-b9bb-7a60caddd98b.mp4

Reply:
Thanks, you're implementation looks correct.

Wondering if this change should happen now.. Maybe doing this in the router is the right place for it? Thoughts?